targets/TARGET_TOSHIBA/TARGET_TMPM46B/sleep.c

```diff
@@ -0,0 +1,33 @@
+/* mbed Microcontroller Library
+ * (C)Copyright TOSHIBA ELECTRONIC DEVICES & STORAGE CORPORATION 2017 All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "sleep_api.h"
+
+void hal_sleep(void)
+{
+    // Set low power consumption mode IDLE
+    CG_SetSTBYMode(CG_STBY_MODE_IDLE);
+    __DSB();
+    // Enter idle mode
+    __WFI();
+}
+
+void hal_deepsleep(void)
+{
+    // deepsleep = sleep because, TMPM46BF10FG does not support the low power
+    // consumption mode configured with the SLEEPDEEP bit in the Cortex-M4 core.
+    // Setting the bit of the system control register is prohibited.
+    hal_sleep();
```

Comment:
if this is the case, then does this mean, deep sleep would not turn of high freq clocks in this case (would it need to be manual here or ?) 

cc @c1728p9

Reply:
@ganesh-ramachandran How many low power modes does this MCU have? Is it just one ? No more further low power oriented ? Like some clocks off, only few peripherals active, etc?

Reply:
@0xc0170, Implemented deepsleep using the below information:
TMPM46BF10FG has three low power consumption modes: IDLE, STOP1 and STOP2. To shift to the low power consumption mode, specify the mode in the system control register and execute the WFI instruction. In this case, execute reset or generate the interrupt to release the mode.
Note 1: The TMPM46BF10FG does not offer any event for releasing the low power consumption mode. Transition to the low power consumption mode by executing the WFE instruction is prohibited.
Note 2: The TMPM46BF10FG does not support the low power consumption mode configured with the SLEEPDEEP bit in the Cortex-M4 core. Setting the <SLEEPDEEP> bit of the system control register is prohibited.
Note 3: Transition to the power consumption mode while the warm-up timer is operating is prohibited.

Reply:
@0xc0170, Please review and confirm the deepsleep changes.